File: code-challenges/Data-Structures/breadthFirstGraph/breadth-first-graph.js
const Graphs = require('../graphs/graphs')
const { Queue } = require('../stacksAndQueues/stacks-and-queues')

class BFT extends Graphs {
  breadthFirst (node) {
    const result = []
    const visited = []
    const nodeQueue = new Queue()
    this.adjList.forEach((value, key) => {
      visited.push({ [key]: false })
    })
    visited[node] = true
    nodeQueue.enqueue(node)

    while (!nodeQueue.isEmpty()) {
      const queuedNode = nodeQueue.dequeue()
      const neighborList = this.adjList.get(queuedNode)
      neighborList.forEach(currNeighbor => {
        if (!visited[currNeighbor]) {
          visited[currNeighbor] = true
          nodeQueue.enqueue(currNeighbor)
        }
      })
      result.push(queuedNode)
    }
    return result
  }
}

module.exports = BFT

File: code_challenges/breadth_first_graph/breadth_first_graph.py
from code_challenges.data_structures.graphs.graphs import Graph
from code_challenges.data_structures.stacks_and_queues.stacks_and_queues import Queue

class BFG(Graph):

    def breadth_first(self, node):
        result, visited, node_queue = [], dict(), Queue()
        for key in self.adj_map.keys():
            visited[key] = False
        visited[node] = True
        node_queue.enqueue(node)

        while not node_queue.isEmpty():
            queued_node = node_queue.dequeue()
            neighbor_list = self.adj_map.get(queued_node)
            for neighbor_map in neighbor_list:
                neighbor = neighbor_map.get('node')
                print(neighbor)
                if not visited[neighbor]:
                    visited[neighbor] = True
                    node_queue.enqueue(neighbor)
            result.append(queued_node)
        return result
